packages/instrumentation-node-cache/src/node-cache.ts

```diff
@@ -0,0 +1,132 @@
+                        'mset',
+                        (args: any[]) => `mset ${args[0].map((entry: NodeCache.ValueSetItem) => entry.key).join(',')}`,
+                        moduleVersion
+                    )
+                    .bind(self)
+            );
+            self._wrap(
+                inst,
+                'ttl',
+                self.patchClassFunction('ttl', (args: any[]) => `ttl ${args[0]} ${args[1]}`, moduleVersion).bind(self)
+            );
+            return inst;
+        }
+        PatchedNodeCache.prototype = origPrototype;
+        PatchedNodeCache.version = (moduleExports as any).version;
+        return PatchedNodeCache;
+    }
+
+    private patchClassFunction(opName: string, toStatement: (...args: any[]) => string, moduleVersion: string) {
+        const self = this;
+        return (originalFunc: Function) => {
+            return function func() {
+                if (self._config.requireParentSpan && trace.getSpan(context.active()) === undefined) {
+                    return originalFunc.apply(this, arguments);
+                }
+
+                const span = self.tracer.startSpan(`node-cache ${opName}`, {
+                    kind: SpanKind.INTERNAL,
+                    attributes: {
+                        [SemanticAttributes.DB_SYSTEM]: 'node-cache',
+                        [SemanticAttributes.DB_OPERATION]: opName,
+                        [SemanticAttributes.DB_STATEMENT]: toStatement(arguments),
+                    },
+                });
+
+                if (self._config.requestHook) {
+                    try {
+                        self._config.requestHook(span, {
+                            moduleVersion,
+                            operation: opName,
+                            args: Object.values(arguments),
+                        });
+                    } catch (err) {
+                        diag.error('node-cache instrumentation: requestHook error', err);
+                    }
+                }
+
+                try {
+                    // Some operations, like "take", use other operation under the hood
+                    // We're only interested in the top level operation, so we use "suppressTracing"
+                    const response = context.with(suppressTracing(context.active()), () =>
+                        originalFunc.apply(this, arguments)
+                    );
+                    if (self._config.responseHook) {
+                        try {
+                            self._config.responseHook(span, response);
+                        } catch (err) {
+                            diag.error('node-cache instrumentation: responseHook error', err);
+                        }
+                    }
+                    return response;
+                } catch (err) {
+                    span.recordException(err);
+                    span.setStatus({
+                        code: SpanStatusCode.ERROR,
+                        message: err.message,
```

Comment:
`err?.message`. can `err` be of type `string` here?

Reply:
fixed